src/Libraries/Microsoft.AspNetCore.Telemetry.Middleware/Logging/LoggingRedactionOptions.cs

```diff
@@ -0,0 +1,104 @@
+    /// </remarks>
+    [Experimental]
+    public HttpRouteParameterRedactionMode RequestPathParameterRedactionMode { get; set; } = DefaultPathParameterRedactionMode;
+
+    /// <summary>
+    /// Gets or sets a map between HTTP path parameters and their data classification.
+    /// </summary>
+    /// <remarks>
+    /// Default set to an empty dictionary.
+    /// If a parameter within a controller's action is not annotated with a data classification attribute and
+    /// it's not found in this map, it will be redacted as if it was <see cref="DataClassification.Unknown"/>.
+    /// If you don't want a parameter to be redacted, mark it as <see cref="DataClassification.None"/>.
+    /// </remarks>
+    [Required]
+    [SuppressMessage("Usage", "CA2227:Collection properties should be read only", Justification = "Options pattern.")]
+    public IDictionary<string, DataClassification> RouteParameterDataClasses { get; set; } = new Dictionary<string, DataClassification>();
+
+    /// <summary>
+    /// Gets or sets a map between request headers to be logged and their data classification.
+    /// </summary>
+    /// <remarks>
+    /// Default set to an empty dictionary.
+    /// That means that no request header will be logged by default.
+    /// </remarks>
+    [Required]
+    [SuppressMessage("Usage", "CA2227:Collection properties should be read only",
+        Justification = "Options pattern.")]
+    public IDictionary<string, DataClassification> RequestHeadersDataClasses { get; set; } = new Dictionary<string, DataClassification>();
+
+    /// <summary>
+    /// Gets or sets a map between response headers to be logged and their data classification.
+    /// </summary>
+    /// <remarks>
+    /// Default set to an empty dictionary.
+    /// That means that no response header will be logged by default.
+    /// </remarks>
+    [Required]
+    [SuppressMessage("Usage", "CA2227:Collection properties should be read only",
+        Justification = "Options pattern.")]
+    public IDictionary<string, DataClassification> ResponseHeadersDataClasses { get; set; } = new Dictionary<string, DataClassification>();
+
+    /// <summary>
+    /// Gets or sets the set of HTTP paths that should be excluded from logging.
+    /// </summary>
+    /// <remarks>
+    /// Any path added to the set will not be logged.
+    /// Paths are case insensitive.
+    /// Default set to an empty <see cref="HashSet{T}"/>.
+    /// </remarks>
+    /// <example>
+    /// A typical set of HTTP paths would be:
+    /// <code>
+    /// ExcludePathStartsWith = new HashSet&lt;string&gt;
+    /// {
+    ///     "/probe/live",
+    ///     "/probe/ready"
+    /// };
+    /// </code>
+    /// </example>
+    [Experimental]
+    [Required]
+    [SuppressMessage("Usage", "CA2227:Collection properties should be read only",
+        Justification = "Options pattern.")]
+    public ISet<string> ExcludePathStartsWith { get; set; } = new HashSet<string>();
```

Comment:
This functionality doesn't seem like it belongs on RedactionOptions in an ideal world. Filtering (in whatever form we think is most appropriate) feels like something that would be available in the base layer without having opted into any redaction. Should I think of this as refactoring is still in progress or we believe this needs to stay here because of various time or design constraints?

Reply:
It's certainly a design point that needs to be resolved. For now it's here because it's an option that the extensions middleware has and the aspnetcore one doesn't. I wouldn't move it to aspnetcore as is, this is an anti-pattern that ends up duplicating routing. I do understand it was done this way because it's easy and config compatible.

We will revisit this detail once we agree on the larger design.

Reply:
What would be a config-friendly way of doing this which is not an anti-pattern?

Reply:
https://github.com/dotnet/aspnetcore/issues/43642 would allow you to apply metadata (like the logging attribute) to arbitrary route paths and be more config friendly. I'll ping them to see how that's coming along.